### PR TITLE
Prepare for 4.28.0 Deployment

### PR DESCRIPTION
**Pull request by sshropshire**
### Summary of changes

 - Update Android Gradle Plugin
 - Tweak Merchant Facing API to improve Kotlin / Java Interop
 - Update Kotlin Version to fix build error

 ### Checklist

 ~- [ ] Added a changelog entry~

- @sshropshire

### CODE REVIEW

File: AmericanExpress/build.gradle [hunk 1]
```diff
@@ -12,8 +12,6 @@ android {
     defaultConfig {
         minSdkVersion rootProject.minSdkVersion
         targetSdkVersion rootProject.targetSdkVersion
-        versionCode rootProject.versionCode
-        versionName rootProject.versionName
```
Comment (sshropshire) [inline AmericanExpress/build.gradle:16]: Android Studio automatically removed these. Apparently they aren't necessary for libraries.

File: BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeError.kt [hunk 2]
```diff
@@ -87,7 +88,12 @@ open class BraintreeError : Parcelable {
             return errors
         }
 
-        internal fun fromGraphQLJsonArray(graphQLErrors: JSONArray?): List<BraintreeError> {
+        /**
+         * @suppress
+         */
+        @JvmStatic
+        @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+        fun fromGraphQLJsonArray(graphQLErrors: JSONArray?): List<BraintreeError> {
```
Comment (sshropshire) [inline BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeError.kt:96]: This is `protected` in Java, so opening up its scope to adhere to semantic versioning. Companion objects cannot have protected methods.

File: BraintreeCore/src/main/java/com/braintreepayments/api/ErrorWithResponse.kt [hunk 3]
```diff
@@ -37,7 +37,11 @@ open class ErrorWithResponse : Exception, Parcelable {
 
     private constructor()
 
-    internal constructor(statusCode: Int, jsonString: String?) {
+    /**
+     * @suppress
+     */
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    constructor(statusCode: Int, jsonString: String?) {
```
Comment (sshropshire) [inline BraintreeCore/src/main/java/com/braintreepayments/api/ErrorWithResponse.kt:44]: Same here. Opening up access.

File: build.gradle [hunk 4]
```diff
@@ -21,7 +21,7 @@ buildscript {
     def roomVersion = System.getenv('CI') ? java7SafeRoomVersion : m1MacSafeRoomVersion
 
     ext.versions = [
-            "kotlin"         : "1.5.32",
+            "kotlin"         : "1.7.0",
```
Comment (sshropshire) [inline build.gradle:24]: Bump Kotlin version to fix build error encountered on latest Android Studio build.

**Review by KunJeongPark — APPROVED** [review]

**Review by demerino — APPROVED** [review]

File: gradle/wrapper/gradle-wrapper.properties [hunk 5]
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.3.3-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.5-bin.zip
```
Comment (scannillo) [inline gradle/wrapper/gradle-wrapper.properties:6]: @sshropshire This bumps us to require Flamingo | 2022.2.1, correct? (Looking [here](https://developer.android.com/studio/releases#android_gradle_plugin_and_android_studio_compatibility))

If so - we should call out the required Android Studio version in the README

Comment (sshropshire) [reply gradle/wrapper/gradle-wrapper.properties:6]: That's a good call. I'll update to make sure, I could see it breaking something.

**Pull request merged**